### ISSUE TITLE
Whitespace control - trim spaces, but not newlines?

### ISSUE THREAD

**Issue by ioleo** [twigphp/Twig#1423]
Example code:

``` html+django
  <pre>
       {{- "My lovely poem:" }}

       {{- "I love twig, oh my, let's make it better, don't ask why!" }}
      {# ^ note this minus sign #}
  </pre>
```

The output html would render as:

`My lovely poem:I love twig, oh my, let's make it better, don't ask why!`

Why? Becouse it not only removes all the spaces, but also newline characters.

Is there a way to "remove all spaces/tabs" until a twig block/character OR newline?

The desired output is:

```
My lovely poem:

I love twig, oh my, let's make it better, don't ask why!
```

I imagine it could be achieved by introduceing another whitespace control sign (that could replace the minus marked in comment) which would work similar to the `-`, but it would stop at newline character.

Or is it already possible somehow? If so - how?

**Comment by etki** [comment]
Having same issues rendering some Yaml templates (foreseeing questions - creating it as internal structure and then serializing it would be even worse)

**Comment by brunoais** [comment]
@fabpot May I try picking this one up?

**Comment by petski** [comment]
For smarty, I've been using https://gist.github.com/rodneyrehm/3013807 for this behavior. The feature in the gist gives even more whitespace control and could be an inspiration for the feature proposed here.

If @ioleo could dust off #1569, that would be great. If not, I'd be happy to invest some time in a PR.

**Comment by ioleo** [comment]
@petski I have since moved to Scala/Java world and no longer have any active PHP projects. It is highly unlikely I will find time.

**Comment by fabpot** [comment]
fixed by #2925
